CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/icon/UriIconDataNGTest.java

```diff
@@ -0,0 +1,224 @@
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * Test class for UriIconData.
+ *
+ * @author sol695510
+ */
+public class UriIconDataNGTest {
+
+    private static MockedStatic<HttpsUtilities> httpsUtilitiesStaticMock;
+    private static MockedStatic<HttpsConnection> httpsConnectionStaticMock;
+    private static MockedStatic<URI> uriStaticMock;
+    private static URI uriMock;
+    private static HttpsConnection httpsConnectionMock;
+    private static HttpsURLConnection httpsURLConnectionMock;
+    private static URL urlMock;
+    private static FileNotFoundException fileNotFoundExceptionMock;
+    private static InputStream inputStreamMock;
+
+    public UriIconDataNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        httpsUtilitiesStaticMock = Mockito.mockStatic(HttpsUtilities.class);
+        httpsConnectionStaticMock = Mockito.mockStatic(HttpsConnection.class);
+        uriStaticMock = Mockito.mockStatic(URI.class);
+        uriMock = Mockito.mock(URI.class);
+        httpsConnectionMock = Mockito.mock(HttpsConnection.class);
+        httpsURLConnectionMock = Mockito.mock(HttpsURLConnection.class);
+        fileNotFoundExceptionMock = Mockito.mock(FileNotFoundException.class);
+        urlMock = Mockito.mock(URL.class);
+        inputStreamMock = Mockito.mock(InputStream.class);
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+        httpsUtilitiesStaticMock.close();
+        httpsConnectionStaticMock.close();
+        uriStaticMock.close();
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+        httpsUtilitiesStaticMock.reset();
+        httpsConnectionStaticMock.reset();
+        uriStaticMock.reset();
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of constructor method with String, of class UriIconData.
+     *
+     * @throws java.lang.Exception
+     */
+    @Test
+    public void testUriIconData_withString() throws Exception {
+        System.out.println("testUriIconData_withString");
+
+        uriStaticMock.when(() -> URI.create(Mockito.any(String.class))).thenReturn(uriMock);
+
+        // If the URI is absolute and has a scheme component.
+        when(uriMock.isAbsolute()).thenReturn(true);
+
+        final UriIconData expResult = new UriIconData("");
+        final UriIconData result = new UriIconData("");
+
+        assertEquals(result, expResult);
+
+        // If the URI is not absolute and does not have a scheme component.
+        when(uriMock.isAbsolute()).thenReturn(false);
+
+        try {
+            final UriIconData instance = new UriIconData("");
+            fail("AssertionError was not thrown.");
+        } catch (final AssertionError er) {
+        }
+    }
```

Comment:
I would separate this into its own unit test (and then have the expected exception as AssertionError, rather than catching it)